### ISSUE TITLE
`--scan` adds the `gradle-enterprise-gradle-plugin` to buildscript classpath again when it already exists

### ISSUE THREAD

**Issue by jprinet** [gradle/gradle#20514]
### Expected Behavior
If a project applies the `gradle-enterprise-gradle-plugin`, adding the `--scan` option shouldn't make a difference since it's already in the build _classpath_.

### Current Behavior
When running the build with `--scan` option, the `gradle-enterprise-gradle-plugin` is directly applied to the build, which in turns modify the build _classpath_ ordering.
This may impact _build cache_ for tasks having the build _classpath_ as an input.

### Context
The `gradle-enterprise-gradle-plugin` is applied directly in `settings.gradle` and defined in an included build.

### Steps to Reproduce 

Here is a project to reproduce the issue:
https://github.com/jprinet/build-classpath-ordering-issue-with-scan

```
./gradlew build
```
https://scans.gradle.com/s/3qbl4vcwkxy3g/build-dependencies?toggled=W1swXSxbMCwwXSxbMCwwLFs0XV1d

```
./gradlew build --scan
```
https://scans.gradle.com/s/6itd7v2z6unk6/build-dependencies?toggled=W1swXSxbMCwwXSxbMCwwLFs1XV1d

With --scan, the _Build Dependencies_ section of a build scan shows direct application of the plugin, although without the option, the plugin is applied through the `enterprise-plugin` defined in the included build.

### Your Environment
Not relevant

**Comment by guylabs** [comment]
See https://gradle.slack.com/archives/C012L0276KF/p1650638139062209 for some more information.

**Comment by bigdaz** [comment]
After further analysis, this issue is a bit broader than the title suggests:

1. Running with `--scan` can result in a newer version of the GE plugin being applied compared to running the same build without `--scan`. This could cause failures publishing a build scan if the Gradle version is newer than the target GE instance.
To see this behaviour, clone https://github.com/bigdaz/duplicate-ge-plugin-application.
- Run `./gradlew build`: [GE plugin `3.8.1` is applied](https://scans.gradle.com/s/65jcisc37o6gm)
- Run `./gradlew build --scan`: [GE plugin `3.10.2` is applied](https://scans.gradle.com/s/x7asmujodpv7k)

2. The issue can arise whenever a custom plugin depends on and applies the GE build scan plugin, whether the plugin is resolved from a repository or consumed via an included build.
To see this behaviour,  checkout the `published-plugin` branch.
- Publish the plugin to maven local: `./gradlew -p build-src publishToMavenLocal`
- Run the same 2 builds as above: same behaviour

3.  The issue is also present for a published plugin that is resolved via `buildscript` block and applied using `apply plugin:`.
To see this behaviour,  checkout the `build-script-block` branch.
- If not already done, publish the plugin to maven local: `./gradlew -p build-src publishToMavenLocal`
- Run the same 2 builds as above: same behaviour

In summary the logic that detects `--scan` and applies the GE plugin fails to detect a pre-existing GE plugin when that the GE plugin is brought in as a transitive dependency of an applied plugin.

**Comment by bigdaz** [comment]
I spent some time trying to find a way to detect if the GE plugin already exists in the graph before auto-applying the version based on `--scan`. In the end, I can't find any non-intrusive mechanism to do this.

The most promising approach I tried was:
1. Instead of merging the auto-applied GE plugin request early ([here](https://github.com/gradle/gradle/blob/79dd695beeb3c272bb9f4053104dee98647ed8db/subprojects/core/src/main/java/org/gradle/configuration/DefaultScriptPluginFactory.java#L114)), wait until after all of the existing plugin requests have been resolved ([here](https://github.com/gradle/gradle/blob/master/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java#L101)).
2. Add a method to [PluginResolution](https://github.com/gradle/gradle/blob/master/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/PluginResolution.java#L27) that indicates whether the resolved plugin includes the GE plugin. We'd need to implement this separately for the various PluginResolution implementations (ArtifactRepository, LocalPluginResolution, ClassPathPluginResolution, etc).
3. If `--scan` is provided on the command-line and no other plugin drags in GE, only then would we resolve the GE plugin and add it to the set of resolved plugins.

**Comment by bigdaz** [comment]
I've raised a separate issue for the case where `--scan` can result in a different version of the GE plugin applied: #22272

**Comment by bigdaz** [comment]
This issue has been superceded by #22272 and #22273
